src/rt/sched/schedulerthread.h

```diff
@@ -168,6 +187,131 @@ namespace verona::rt
       }
     }
 
+    /**
+     * Mute a set of cowns. This will add the cowns to the mute set of the
+     * mutor.
+     */
+    void mute(T** cowns, size_t count)
+    {
+      auto it = mute_map.find(mutor);
+      if (it == mute_map.end())
+      {
+        auto* set = ObjectMap<T*>::create(alloc);
+        it = mute_map.insert(alloc, std::make_pair(mutor, set)).second;
+      }
+      else
+      {
+        mutor->weak_release(alloc);
+      }
+      auto& mute_set = *it.value();
+
+      for (size_t i = 0; i < count; i++)
+      {
+        auto* cown = cowns[i];
+
+        if (
+          (state == ThreadState::PreScan) || (state == ThreadState::Scan) ||
+          (state == ThreadState::AllInScan))
+        { // Messages in this cowns queue must be scanned.
+          cown->schedule();
+          continue;
+        }
+
+        Systematic::cout() << "Mute " << cown << std::endl;
+        auto bp = cown->backpressure.load(std::memory_order_relaxed);
+        bp.set_muted();
+        cown->backpressure.store(bp, std::memory_order_release);
+        T::acquire(cown);
+        mute_set.insert(alloc, cown);
+      }
+
+      alloc->dealloc(cowns, count * sizeof(T*));
+    }
+
+    /**
+     * Unmute all mute sets where the mutor is in a state that triggers
+     * unmuting. If `force` is true, then all mute sets in the map will be
+     * unmuted.
+     */
+    template<bool force = false>
+    void mute_map_scan()
+    {
+      // Handle unmute queue first.
+      while (true)
```

Comment:
Instead of `while(true)`, I'd rather have something like:
```
while ((auto *msg = unmute_q...) != nullptr)
{
  ...
}
```

It's clear what the exit condition is from the first line. I wouldn't have to dig the code to find out...

Reply:
I find using an assignment as an expression very confusing when it returns the newly assigned value. But either way, I can't get the compiler to accept either of the following:
```cpp
error: expected ')'
      while ((auto* msg = unmute_q.dequeue(alloc, notify)) != nullptr)
                    ^
```
```cpp
error: expected ')'
      while (auto* msg = unmute_q.dequeue(alloc, notify); msg != nullptr)
                                                        ^
```

Reply:
Sorry, that'd be without parenthesis, but that's not good either and is super confusing. Ignore this for now.